core/src/main/java/hudson/logging/LogRecorder.java

```diff
@@ -206,6 +203,14 @@ public void disable() {
         }
 
     }
+    
+    private static class TargetComparator implements Comparator<Target> {
+
+        @Override
+        public int compare(Target left, Target right) {
+            return right.getName().length() - left.getName().length();
+        }
+    }
```

Comment:
Well logging systems generally are sensitive to this class of bugs: you have to be very careful to not run any potentially novel classes in the course of handling a log event.